interface/ceed-basis.c

```diff
@@ -657,30 +728,46 @@ int CeedBasisReferenceCopy(CeedBasis basis, CeedBasis *basis_copy) {
 **/
 int CeedBasisView(CeedBasis basis, FILE *stream) {
   int ierr;
-
-  if (basis->tensor_basis) {
-    fprintf(stream, "CeedBasis: dim=%d P=%d Q=%d\n", basis->dim, basis->P_1d,
-            basis->Q_1d);
-    ierr = CeedScalarView("qref1d", "\t% 12.8f", 1, basis->Q_1d, basis->q_ref_1d,
-                          stream); CeedChk(ierr);
-    ierr = CeedScalarView("qweight1d", "\t% 12.8f", 1, basis->Q_1d,
-                          basis->q_weight_1d, stream); CeedChk(ierr);
-    ierr = CeedScalarView("interp1d", "\t% 12.8f", basis->Q_1d, basis->P_1d,
-                          basis->interp_1d, stream); CeedChk(ierr);
-    ierr = CeedScalarView("grad1d", "\t% 12.8f", basis->Q_1d, basis->P_1d,
-                          basis->grad_1d, stream); CeedChk(ierr);
-  } else {
-    fprintf(stream, "CeedBasis: dim=%d P=%d Q=%d\n", basis->dim, basis->P,
-            basis->Q);
+  switch (basis->basis_space) {
+  case 1: // H^1 discretization
```

Comment:
Instead of magic numbers, this should be an enum

Reply:
Yeah, agreed. Can we add enum to the basis constructor? This is something like `tensor_basis` we have it already but it is not 0,1 only. We have L^2, and H(curl) as well. I set 0,1,2,3 for L^2, H^1, H(div), H(curl) (last one not implemented)

Reply:
I was thinking something like this 
https://github.com/CEED/libCEED/blob/7ed177db354372d7a27bd4ae30dbb2e96a56ab87/include/ceed/ceed.h#L490

Reply:
Ah, nice. So with that I think we need to add a new parameter to the basis constructor, right? Parameter like `CeedFESpace`.

Reply:
Which basis constructor? I thought our current basis constructors all specify the space in their names?

Reply:
Yeah, the space are specified in their name. In the link that you shared `CeedElemTopology` is an input for basis constructor. If we need something like that we should add it to the input of the constructor, right?or we don't need to.!?

Reply:
I think we don't need to added as input.Let me check.